openedx/core/djangoapps/programs/tests/test_tasks.py

```diff
@@ -0,0 +1,244 @@
+        """
+        Ensure the function is making the right API calls based on inputs
+        """
+        student = UserFactory()
+        ClientFactory.create(name='programs')
+        api_config = self.create_programs_config(
+            internal_service_url='http://foo',
+            api_version_number=99,
+        )
+        mock_get_id_token.return_value = 'test-token'
+
+        api_client = tasks.get_api_client(api_config, student)
+        self.assertEqual(mock_get_id_token.call_args[0], (student, 'programs'))
+        self.assertEqual(api_client._store['base_url'], 'http://foo/api/v99/')  # pylint: disable=protected-access
+        self.assertEqual(api_client._store['session'].auth.token, 'test-token')  # pylint: disable=protected-access
+
+
+class GetCompletedCoursesTestCase(TestCase):
+    """
+    Test the get_completed_courses function
+    """
+
+    def make_cert_result(self, **kwargs):
+        """
+        Helper to create dummy results from the certificates API
+        """
+        result = {
+            'username': 'dummy-username',
+            'course_key': 'dummy-course',
+            'type': 'dummy-type',
+            'status': 'dummy-status',
+            'download_url': 'http://www.example.com/cert.pdf',
+            'grade': '0.98',
+            'created': '2015-07-31T00:00:00Z',
+            'modified': '2015-07-31T00:00:00Z',
+        }
+        result.update(**kwargs)
+        return result
+
+    @mock.patch('openedx.core.djangoapps.programs.tasks.get_certificates_for_user')
+    def test_get_completed_courses(self, mock_get_certs_for_user):
+        """
+        Ensure the function correctly calls to and handles results from the
+        certificates API
+        """
+        student = UserFactory(username='test-username')
+        mock_get_certs_for_user.return_value = [
+            self.make_cert_result(status='downloadable', type='verified', course_key='downloadable-course'),
+            self.make_cert_result(status='generating', type='prof-ed', course_key='generating-course'),
+            self.make_cert_result(status='unknown', type='honor', course_key='unknown-course'),
+        ]
+
+        result = tasks.get_completed_courses(student)
+        self.assertEqual(mock_get_certs_for_user.call_args[0], (student.username, ))
+        self.assertEqual(result, [
+            {'course_id': 'downloadable-course', 'mode': 'verified'},
+            {'course_id': 'generating-course', 'mode': 'prof-ed'},
+        ])
+
+
+class GetCompletedProgramsTestCase(TestCase):
+    """
+    Test the get_completed_programs function
+    """
+
+    @httpretty.activate
+    def test_get_completed_programs(self):
+        """
+        Ensure the correct API call gets made
+        """
+        test_client = EdxRestApiClient('http://test-server', jwt='test-token')
+        httpretty.register_uri(
+            httpretty.POST,
+            'http://test-server/programs/complete/',
+            body='[1, 2, 3]',
```

Comment:
This body should be updated to match what programs returns. Or, we should update programs so that it returns a body of this form.